### PR TITLE
Ades/claim curve

### PR DESCRIPTION
**Pull request by sennett**
# [Claim LDO rewards](https://app.shortcut.com/oazo-apps/story/3895/add-a-banner-for-claiming-lido-rewards)
  
Adds functionality to claim LDO rewards as part of CRVV1ETHSTETH-A ilk.

image

## Changes 👷‍♀️

For Ilks that support rewards (only CRVV1ETHSTETH-A right now), the box is displayed.  The user can click the button to claim their rewards.

Tech notes:

- New integrations with smart contracts:
  - Lido Curve liquidity farming reward, taken from https://docs.lido.fi/deployed-contracts
  - LDO ERC20, also from https://docs.lido.fi/deployed-contracts
  - Maker Cropjoin from makerdao chainlog
- inside `/features/bonus`:
  - New pipe bonusPipe, with a makerprotocol-specific adapter that handles the blockchain stuff.  Pipe is independent of  manage borrow pipe.
  - BonusContainer that consumes the pipe and provides the UI.
- Updates erc20 abi to include decimals (taken from second google result https://gist.github.com/veox/8800debbf56e24718f9f483e1e40c35c)
  
## How to test 🧪

- find a CRVV1ETHSTETH-A vault from https://tracker-vaults.makerdao.network/collateral/CRVV1ETHSTETH-A
- click through to ethx and cmd-f for `open`, then find the CDP ID.
- use hardhat and steal the proxy for the CDP ID using hardhat task `npx hardhat get-proxy --proxy <proxy_addr> --network local` where <proxy_addr> is the proxy of the CDP ID.  Your first Hardhat wallet will now have the proxy.
- oasis-borrow at hardhat and use the first hardhat address.
- Navigate to the CDP ID (it should be a CRVV1ETHSTETH-A position)
- You should be able to see the amount of unclaimed LDO, claim the LDO, and see the same amount get transferred to the wallet.

### CODE REVIEW

**Review by BackToTheCode — DISMISSED** [review]

File: blockchain/calls/proxyActions/adapters/ManagerlessProxyActionsAdapter.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
 import { ContextConnected } from '../../../network'
```
[the remark this answers is not in the document]
Comment (sennett) [reply blockchain/calls/proxyActions/adapters/ManagerlessProxyActionsAdapter.ts:10]: Good idea - have added.

File: features/bonus/bonusPipe.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  claimTxnState?: ClaimTxnState
+}
+
+export type BonusAdapter = {
+  bonus$: Observable<Bonus>
+  claimAll$: Observable<(() => Observable<ClaimTxnState>) | undefined>
+}
+
+export function createBonusPipe$(
+  createbBonusAdapter: (cdpId: BigNumber) => BonusAdapter,
```
[the remark this answers is not in the document]
Comment (sennett) [reply features/bonus/bonusPipe.ts:33]: Done.

File: features/bonus/BonusContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+export function BonusContainer(props: BonusContainerProps) {
+  const { bonus$ } = useAppContext()
+  const [bonusViewModel] = useObservable(bonus$(props.cdpId))
+  const { t } = useTranslation()
+
+  if (bonusViewModel) {
+    const { bonus, claimAll, claimTxnState } = bonusViewModel
+    const bonusInstructionSnippet = bonus.amountToClaim.eq(0)
```
[the remark this answers is not in the document]
Comment (sennett) [reply features/bonus/BonusContainer.tsx:22]: Yeah I don't like them either - I have updated.

File: blockchain/calls/cropper.ts [hunk 4]
```diff
@@ -1,7 +1,81 @@
...
+  prepareArgs: () => [],
+  postprocess: (result: any) => {
+    return new BigNumber(result) // crops per gem    [bonus decimals * ray / wad]
+  },
+}
+
+export const cropperBonusTokenAddress: CallDef<{ ilk: string }, string> = {
+  call: ({ ilk }, { web3, joins }) => {
+    const join = joins[ilk]
+    const contract = new web3.eth.Contract((mcdCropJoinAbi as any).default, join)
```
[the remark this answers is not in the document]
Comment (sennett) [reply blockchain/calls/cropper.ts]: Yup done.

**Review by BackToTheCode — APPROVED** [review]

**Pull request merged**